x/dispensation/types/querier.pb.go

```diff
@@ -0,0 +1,544 @@
+// Code generated by protoc-gen-gogo. DO NOT EDIT.
+// source: sifnode/dispensation/v1/querier.proto
+
+package types
+
+import (
+	fmt "fmt"
+	github_com_cosmos_cosmos_sdk_types "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/gogo/protobuf/gogoproto"
+	proto "github.com/gogo/protobuf/proto"
+	io "io"
+	math "math"
+	math_bits "math/bits"
+)
+
+// Reference imports to suppress errors if they are not otherwise used.
+var _ = proto.Marshal
+var _ = fmt.Errorf
+var _ = math.Inf
+
+// This is a compile-time assertion to ensure that this generated file
+// is compatible with the proto package it is being compiled against.
+// A compilation error at this line likely means your copy of the
+// proto package needs to be updated.
+const _ = proto.GoGoProtoPackageIsVersion3 // please upgrade the proto package
+
+type QueryRecordsByRecipientAddr struct {
+	Address string `protobuf:"bytes,1,opt,name=address,proto3" json:"address,omitempty"`
+}
+
+func (m *QueryRecordsByRecipientAddr) Reset()         { *m = QueryRecordsByRecipientAddr{} }
+func (m *QueryRecordsByRecipientAddr) String() string { return proto.CompactTextString(m) }
+func (*QueryRecordsByRecipientAddr) ProtoMessage()    {}
+func (*QueryRecordsByRecipientAddr) Descriptor() ([]byte, []int) {
+	return fileDescriptor_5362b46faa94baa8, []int{0}
+}
+func (m *QueryRecordsByRecipientAddr) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *QueryRecordsByRecipientAddr) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_QueryRecordsByRecipientAddr.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *QueryRecordsByRecipientAddr) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_QueryRecordsByRecipientAddr.Merge(m, src)
+}
+func (m *QueryRecordsByRecipientAddr) XXX_Size() int {
+	return m.Size()
+}
+func (m *QueryRecordsByRecipientAddr) XXX_DiscardUnknown() {
+	xxx_messageInfo_QueryRecordsByRecipientAddr.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_QueryRecordsByRecipientAddr proto.InternalMessageInfo
+
+func (m *QueryRecordsByRecipientAddr) GetAddress() string {
+	if m != nil {
+		return m.Address
+	}
+	return ""
+}
+
+type QueryRecordsByDistributionName struct {
+	DistributionName string                                  `protobuf:"bytes,1,opt,name=distribution_name,json=distributionName,proto3" json:"distribution_name,omitempty"`
+	Status           github_com_cosmos_cosmos_sdk_types.Uint `protobuf:"bytes,2,opt,name=status,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Uint" json:"status" yaml:"status"`
```

Comment:
status was defined as a type earlier . Can we go back to that or is this a limitation caused by proto ,something like .
type DispensationStatus uint64 

so that we can use constants instead fo using numbers . like
Pending DispensationStatus = 1

Reply:
We can use a similar pattern as the on used for DistributionType , in types.pb.go

Reply:
DispensationStatus would've been a better name, but I see it is a ClaimStatus enum within the dispensation module now.